src/main/java/com/commercetools/sync/commons/helpers/BaseReferenceResolver.java

```diff
@@ -68,7 +67,8 @@ protected static String getKeyFromResourceIdentifier(@Nonnull final ResourceIden
                                                          final boolean allowUuidKeys)
         throws ReferenceResolutionException {
         final String key = resourceIdentifier.getId();
-        validateKey(key, allowUuidKeys, UNSET_ID_FIELD);
+        validateKey(key, allowUuidKeys, "The value of 'id' field of the Resource Identifier is blank "
+            + "(null/empty).");
```

Comment:
do you really need :(null/empty)

Reply:
yes, to explicitly explain what is blank in code terms.